scanpipe/pipes/__init__.py

```diff
@@ -192,8 +193,12 @@ def update_or_create_package(project, package_data, codebase_resources=None):
     else:
         package = DiscoveredPackage.create_from_data(project, package_data)
 
-    if package and codebase_resources:
-        package.add_resources(codebase_resources)
+    if package:
+        if datasource_id and datasource_id not in package.datasource_ids:
+            package.update(datasource_ids=[datasource_id])
```

Comment:
Is there a risk of losing data here? Should the datasource_id be added to the list instead?

Reply:
Adding to list instead, thanks!